axiom/nr_segment.c

```diff
@@ -852,6 +882,17 @@ void nr_segment_destroy_tree(nr_segment_t* root) {
       root, (nr_segment_iter_t)nr_segment_destroy_children_callback, NULL);
 }
 
+void nr_segment_end_tree(nr_segment_t* curr, nr_segment_t* root) {
+  nr_segment_t* next;
+  if (NULL == curr || root == NULL) {return;}
+  next = curr->parent;
+  while (curr != root) {
+    nr_segment_end(&curr);
+    curr = next;
+    next = curr->parent;
```

Comment:
is `next` aka `curr->parent` always guaranteed to be non-`null`?

Reply:
(and `curr` for that matter as well)

Reply:
They should be guaranteed non-null. At the very least, if they are null, we have a memory leak, as that would indicate that the segment tree is malformed. For safety (if this PR is chosen over https://github.com/newrelic/newrelic-php-agent/pull/801), I can add a null check with an error log.

Reply:
I think adding a `null` check is a good idea if this PR is selected over the other.